### PR TITLE
[website] Update portal for v2

### PR DESCRIPTION
**Pull request by BMillman19**
## Description

* Updated all `@0xproject/*` dependencies to rc versions
* Refactored internal `Order` type to use the canonical `SignedOrder` type
* Added old version of `0x.js` as a dependency to continue support for token registry
* Fixed a bug with `@0xproject/contract-wrappers` where the wrong error message was being thrown when a contract was found to be not deployed on a network

## Testing instructions

TBD - dogfood link goes here

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

*   [ ] Prefix PR title with `[WIP]` if necessary.
*   [ ] Prefix PR title with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [ ] Add tests to cover changes as needed.
*   [ ] Update documentation as needed.
*   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by tomhschmidt** [comment]
Are we still removing or erroring order generation and fill features / is that a separate PR?

**Comment by BMillman19** [comment]
@tomhschmidt it works with v2

File: packages/website/ts/components/generate_order/generate_order_form.tsx [hunk 1]
```diff
@@ -308,39 +317,43 @@ export class GenerateOrderForm extends React.Component<GenerateOrderFormProps, G
...
                 this.props.userAddress,
                 hashData.makerFee,
                 hashData.takerFee,
                 hashData.feeRecipientAddress,
-                ecSignature,
+                signature,
                 this.props.tokenByAddress,
                 hashData.orderSalt,
             );
+            console.log(order);
```
Comment (BMillman19) [inline packages/website/ts/components/generate_order/generate_order_form.tsx]: remove

File: packages/website/ts/utils/order_parser.ts [hunk 2]
```diff
@@ -28,6 +28,48 @@ export const orderParser = {
...
+        const convertedSignedOrder = convertOrderStringFieldsToBigNumber(signedOrder);
+        const result = {
+            ...order,
+            signedOrder: convertedSignedOrder,
+        };
+        return result;
     },
 };
+
+// TODO: consolidate this function with that in typeConverters in @0xproject/connect
```
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/website/ts/utils/order_parser.ts]: I'd prefer we move them to `order-utils`

File: packages/website/ts/blockchain.ts [hunk 3]
```diff
@@ -402,22 +369,27 @@ export class Blockchain {
...
-        cancelTakerTokenAmount: BigNumber,
-    ): Promise<void> {
-        await this._contractWrappers.exchange.validateCancelOrderThrowIfInvalidAsync(order, cancelTakerTokenAmount);
+        // TODO: add validation here
+        // we can use OrderValidationUtils here?
+        // await this._contractWrappers.exchange.validateFillOrderThrowIfInvalidAsync(
+        //     signedOrder,
+        //     fillTakerTokenAmount,
+        //     takerAddress,
+        // );
```
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/website/ts/blockchain.ts]: BTW: Just re-added the validation methods to `contract-wrappers`: https://github.com/0xProject/0x-monorepo/pull/1054

File: packages/website/ts/utils/utils.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
+import { ContractWrappersError } from '@0xproject/contract-wrappers';
+import { assetDataUtils, OrderError } from '@0xproject/order-utils';
 import { constants as sharedConstants, Networks } from '@0xproject/react-shared';
-import { ECSignature, Provider } from '@0xproject/types';
+import { ECSignature, ExchangeContractErrs } from '@0xproject/types';
 import { BigNumber } from '@0xproject/utils';
 import { Web3Wrapper } from '@0xproject/web3-wrapper';
 import * as bowser from 'bowser';
 import deepEqual = require('deep-equal');
+import { Provider } from 'ethereum-types';
```
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/website/ts/utils/utils.ts]: removed.

File: packages/website/ts/blockchain.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/website/ts/blockchain.ts]: done.

File: packages/website/ts/blockchain.ts [hunk 5]
```diff
@@ -746,28 +722,28 @@ export class Blockchain {
             tradeHistoryStorage.addFillToUser(this._userAddressIfExists, this.networkId, fill);
         }
     }
-    private async _convertDecodedLogToFillAsync(
-        decodedLog: LogWithDecodedArgs<LogFillContractEventArgs>,
-    ): Promise<Fill> {
```
[the remark this answers is not in the document]
Comment (fabioberger) [reply packages/website/ts/blockchain.ts]: What do you mean by this?

**Pull request merged**